### DISCUSSION TITLE
const incompatible not working //resolved

### DISCUSSION THREAD

**Discussion by Roskaiu** [nftchef/art-engine#265] [Help + Q&A]
![image](https://user-images.githubusercontent.com/104800157/168204212-68762617-d4f4-4435-a513-549231b31ef4.png)

This would not be working for me. Can someone help me find the error?

**Comment by Roskaiu** [comment]
we have tried different methods to make it work without success

**Comment by PhosphoriaArtist** [comment]
Hey, what exactly are you trying to make incompatible with what?
In this example you've given you are saying if the "Hat" folder is chosen (in your layer config), then do not go down the "BAOAhat". "FeltHat", "FishingHat" etc folders later on in the layer config. 
I don't know your setup but it seems like 'hat' is a very general term that contains all the other hats. What's the structure of your folders or layer config?

**Comment by Roskaiu** [comment]
Hello. Something like this is my folder structure.

When a "Hat" comes out, I don't want the "LaserBeam" to come out.

Thank you very much for your reply.
               
               
  
¦ 
¦   +---Background
¦   ¦       Back1#10.png
¦   ¦       Back2#90.png
¦   ¦       
¦   +---Hair Style
¦   ¦	  	Curly#10.png
¦   ¦		Short#10.png
+---Red
¦   +---Facial Hair
¦   ¦       Side Burns#10.png
¦   ¦       NONE#90.png
¦   ¦       
¦   +---Hair Style
¦   ¦		Curly#10.png
¦   ¦		Short#10.png
¦   ¦
¦   +---Facial Hair
¦   ¦       Side Burns#10.png
¦   ¦       NONE#90.png
¦   ¦       
¦   +---HatBAOAHat
¦   ¦             FeltHat#10.png
¦   ¦             FishingHat#10.png
¦   ¦             HeadBand#10.png
¦   ¦             Helmet#20.png
¦   ¦             LSD_420Helmet#10.png
¦   ¦             LSD_420inEye#40.png 
¦   ¦        
¦    +---LaserEye
¦   ¦             GlowingGreenLaserEyes#25.png
¦   ¦               PurpleLaserEyes#25.png
¦   ¦               SolidGreenLaserEyes#25.png
¦   ¦               YellowLaserEyes#25.png

**Comment by PhosphoriaArtist** [comment]
Ah! so you want 

const incompatible = {
//if HATBAOAHat folder is chosen, do NOT choose LaserEye.
HatBAOAHat: ["LaserEye"],
};
Format is:
`<Folder name that is chosen earliest in layer config>: [<"stuff that shouldn't be chosen after that">],`

You could also use nested folders to ensure they are never chosen together, like:
Folders:
Headstuff
¦ +Hats#50
¦ ¦ HatStuff
¦ +LaserEye#50
 ¦ ¦ LaserStuff
   
So the generator will go down Heastuff as a main layer, then pick either Hats or LaserEye to go down.

**Comment by jjclay** [comment]
The key can only be listed once, with multiple incompatibilities between the brackets, like this:

"FeltHat": ["GreenLaserEyes", "PurpleLaserEyes"],

**Comment by Roskaiu** [comment]
> The key can only be listed once, with multiple incompatibilities between the brackets, like this:
> 
> "FeltHat": ["GreenLaserEyes", "PurpleLaserEyes"],

Works

**Comment by cryptofannatic** [comment]
Can someone please help with this - https://github.com/nftchef/art-engine/discussions/232 i think I might be facing a similar issue.
